agent/server.go

```diff
@@ -105,6 +108,34 @@ func (r *server) UpdateTimeline(ctx context.Context, req *pb.UpdateRequest) (*pb
 	return &pb.UpdateResponse{}, nil
 }
 
+// Stop stops the grpc server and any additional http servers.
+// TODO: modify Stop to return error
+func (r *server) Stop() {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	if err := r.stopHTTPServers(ctx); err != nil {
+		log.WithError(err).Error("All HTTP servers failed to shutdown.")
```

Comment:
Technically not true about `all` as the method might return errors for a subset.